JIT.cpp

```diff
@@ -20,6 +20,16 @@ void JIT::AddModule(std::unique_ptr<llvm::Module> module)
                                        orc::ThreadSafeModule(std::move(module), m_ctx)));
 }
 
+bool JIT::ContainsSymbol(llvm::StringRef symbol_name)
+{
+    auto symbol = m_execution_session.lookup({ &m_execution_session.getMainJITDylib() },
+                                             m_mangler(symbol_name.str()));
+    if(symbol)
```

Comment:
clang-format?  (space after `if` missing).  I would fold this:

```
if (auto symbol = m_execution_session.lookup({&m_execution_session.getMainJITDylib()},
                                             m_mangler(symbol_name.str()))
  return true;
```

Any idea how expensive it is?  Could you write a test case where you create a 100000 functions and perform a lookup?  Might be worth it do use a `DenseMap` side table lookup.

Reply:
You can't wrap it because I want to eat the error. I'll look into speed.

Reply:
I don't even use this function, deleting it.